enable/examples/demo/enable/sierpinski_zoom.py

```diff
@@ -0,0 +1,207 @@
+        completion of the full recursive call.
+        """
+        size = self.base_width/4 * (1/2)**(n - 1)
+        if n == self.iterations:
+            return
+        else:
+            # find top left corners of next 3 inverted triangles relative to
+            # the top left corner of the current one (ie point)
+            #
+            # In the diagram below, X is point, and the ? represent the next
+            # points to make recursive calls at
+            #
+            #       /\
+            #     ?/__\
+            #    X/_\/_\
+            #    /\    /\
+            #  ?/__\ ?/__\
+            #  /_\/_\/_\/_\
+            #
+
+            rel_to_point_locs = (size/2)*np.array([
+                [-1, -SQRT3],
+                [1, SQRT3],
+                [3, -SQRT3]
+            ])
+            # absolute location of those next points
+            abs_points = point + rel_to_point_locs
+
+            for point in abs_points:
+                self.add_triangle_to_path(
+                    path,
+                    point,
+                    size
+                )
+                self.sierpinski(
+                    path, point, n+1
+                )
+
+    def add_triangle_to_path(self, path, point, size):
+        """ Adds an inverted triangle to the input path, of side length size,
+        using the input point as the upper left corner of the triangle.
+        """
+        x, y = point
+        path.move_to(x, y)
+        path.line_to(x + size, y)
+        path.line_to(x + .5 * size, y - (SQRT3 / 2) * size)
+        path.line_to(x, y)
+
+
+class Viewer(HasTraits):
+
+    iterations = Range(0, 'max_iters')
+
+    triangle = Instance(AbstractOverlay)
+
+    comp = Instance(Component)
+
+    base_width = Int(500)
+
+    max_iters = Property(observe="base_width, viewport.zoom")
+
+    viewport = Instance(Viewport)
+
+    def _get_bounds(self):
+        return [self.base_width, self.base_width*(SQRT3/2)]
+
+    def _get_max_iters(self):
+        factor = 1
+        if self.viewport:
+            factor = self.viewport.zoom
+        return int(np.log(4/(factor*self.base_width))/np.log(.5) + 1)
+
+    def _comp_default(self):
+        canvas = Canvas(draw_axes=True, bgcolor="gray")
+        self.triangle = SierpinskiTriangle(
+            canvas,
+            position=[0.0, 0.0],
+            bounds=[self.base_width, self.base_width*(SQRT3/2)],
+            iterations=self.iterations,
+            max_iters=self.max_iters,
+            base_width=self.base_width,
+        )
+        canvas.overlays.append(self.triangle)
+
+        self.viewport = Viewport(
+            component=canvas, enable_zoom=True, stay_inside=True
+        )
+        self.viewport.view_position = [0, 0]
+        self.viewport.view_bounds = [
+            self.base_width, self.base_width*(SQRT3/2)
+        ]
+        self.viewport.tools.append(
+            ViewportPanTool(self.viewport, drag_button="right")
+        )
```

Comment:
`Viewport` is very strange... It has a `zoom_tool` trait so zoom is handled differently from other tools, eg `ViewportPanTool`.  Additionally, with this line:
https://github.com/enthought/enable/blob/7f3231d14d118b7a7966af0cd4f0c257340569b9/enable/viewport.py#L97-L98
we create a `ViewportZoomTool` instance by default if `enable_zoom` is True, passing in `self` (the viewport) as the component.  That is all fine and good, but I was struggling to see how I could pass in a zoom_tool other than this... If I want to pass in my own zoom tool when instantiation a viewport, eg:
```
my_zoom_tool = ViewportZoomTool(...)
my_viewport = Viewport(..., zoom_tool=my_zoom_tool)
```
I was running into a chicken and egg scenario as I needed the `my_viewport` instance to pass in as the component for the `ViewportZoomTool`, but it isn't defined when I define `my_zoom_tool`.  If I try to switch the order:
```
my_viewport = Viewport(..., zoom_tool=my_zoom_tool)
my_zoom_tool = ViewportZoomTool(...)
```
Then the above line gets run so I end up using the default zoom tool, not `my_zoom_tool`.  Thus I resorted to just setting the trait I wanted to change on the default zoom tool that got created, rather than trying to instantiate my own.
This all works as I want it to.  However, I do not understand when a user would want / be able to set the `zoom_tool` trait.  Maybe if the `ViewportZoomTool` was working with a second `Viewport` or something?
However, as far as I understand these are always with a `Canvas` (that very well may not actually be true, but there isn't documentation on this / there aren't many examples) , and `Canvas` has the comment:
https://github.com/enthought/enable/blob/7f3231d14d118b7a7966af0cd4f0c257340569b9/enable/canvas.py#L34-L35
so that doesn't seem possible either.


Additionally, `ViewportZoomTool` has traits `max_zoom` / `min_zoom`, but its base class `BaseZoomTool` has traits `max_zoom_in_factor` and `min_zoom_in_factor` which appear to have the same purpose.  

There are still many things about `Viewport` and `ViewportZoomTool` I do not fully understand / I was unable to get to work how I expected.  e.g. the `{horizontal/vertical}_anchor` traits, `stay_inside` / `view_bounds`.


In any case, this example effectively works as I want it to.  Perhaps I should open some issues about these things / an issue about adding documentation on how to use zoom in enable generally.  I imagine enable zoom is a pretty niche thing as chaco handles zoom on its own, but even so.